sdk/keyvault/keyvault-keys/src/cryptography/rsaCryptographyProvider.ts

```diff
@@ -191,8 +188,12 @@ export class RsaCryptographyProvider implements CryptographyProvider {
     RS512: "SHA512"
   };
 
-  private ensureValid() {
-    if (this.key && this.key.kty! !== "RSA" && this.key.kty! !== "RSA-HSM") {
+  private ensureValid(): void {
+    if (
+      this.key &&
+      this.key.kty?.toUpperCase() !== "RSA" &&
```

Comment:
Nit/Q: Why uppercase here and lowercase for AES?

Reply:
Good Q - the actual key types are RSA / RSA-HSM, and oct / oct-HSM which is probably why I did it that way. But now I think that's confusing and unnecessary so I'll change AES to uppercase..